runtime/src/mem/jemalloc/mem-jemalloc.c

```diff
@@ -28,15 +28,40 @@
 #include "chpl-comm.h"
 #include "chpl-mem.h"
 #include "chplmemtrack.h"
+#include "chplsys.h"
+#include "chpl-topo.h"
 #include "chpltypes.h"
 #include "error.h"
 
-static struct shared_heap {
+
+//
+// Information about shared heaps gotten from the comm layer.
+//
+typedef struct shared_heap {
   void* base;
   size_t size;
   size_t cur_offset;
   pthread_mutex_t alloc_lock;
-} heap; // static, will be "zero" initialized automatically
+} shared_heap_t;
+
+static shared_heap_t* heaps;
+static int num_heaps;
+
+static inline int get_num_heaps(void) {
+  return num_heaps;
```

Comment:
I could.  Note that that'll necessitate an "is this set?" check on every access and also either an implicit assertion that the first access precedes all parallelism or a parallel-unsafe initialization at least for non-POSIX systems where we don't have pthread_once().  (I'm sensitive to this issue because I just went round a bit with another reviewer a few days ago about a variable initialization I'd done under pthread_once() as a future-proofing step, even though I knew that in current code the initialization would definitely occur before any concurrency.)